### PR TITLE
Enhance Documentation in the Project #76

### LINKED ISSUE #76: Enhance Documentation in the Project
## 📖 Description
- We want to improve the documentation organization in the project, and for this, we need the following:
  - Create a new `docs` directory
  - Inside the new dir create a `contribution` directory 
  - Place the Contribution guidelines inside of it
- Another improvement we want is to create a template for the issues we will make in the future, this can be done by adding a template.md inside the .github directory, follow the pull request MD file as an example
   - Inside the .github, create a new directory called `templates` and place both the pull request and the new issue templates in it. 
   - Use the descriptions from the ODHack9 issues in this repository as references to create the template. Feel free to add any additional points you think would be helpful!

## 📦 Deliverable
- `docs` directory created
- Contribution guidelines moved to the correct place
- Issue template created
- Both templates moved to the correct place 

## ⚠️ Notes
- This issue is specifically for newcomers, those just starting out with OnlyDust and open-source contributions. If you're an experienced developer, please leave this issue to someone who is new. There are other issues better suited for seasoned developers and OSS contributors!

### PR DESCRIPTION
**Pull request by josephpdf**
### 📝 Summary 
- Closes #76 
### 🔄 Changes Made
The changes made were:
- Created a new directory named `docs` with a subdirectory called `contribution`.
- Moved the `CONTRIBUTION.md` file to the `contribution` folder.
- Created a `templates` directory inside `.github`.
- Moved the `PULL_REQUEST_TEMPLATE.md` file to the `templates` directory.
- Added a new file named `ISSUE_TEMPLATE.md` with various sections, using examples from the issues in ODHack9 as references.
### 🔧 Tests Results
- 
### 🔜 Next Steps
- 

## Summary by CodeRabbit

- **Documentation**
	- Updated the link to the contribution guidelines in the README for improved accessibility.
	- Introduced a new structured issue template to guide users in submitting detailed issue reports, enhancing clarity and organization.

### CODE REVIEW

File: .github/templates/TEMPLATE.md
Comment (danielcdz) [inline .github/templates/TEMPLATE.md:1]: Can you rename this file to `ISSUE_TEMPLATE`

**Comment by danielcdz** [comment]
@josephpdf Can you translate the PR description steps you added to English, please?

File: .github/templates/TEMPLATE.md
Comment (josephpdf) [reply .github/templates/TEMPLATE.md:1]: I have already changed the file name.

**Comment by josephpdf** [comment]
> @josephpdf Can you translate the PR description steps you added to English, please?

I was wrong, but I already changed the language.

File: docs/contribution/CONTRIBUTION.md
Comment (coxmars) [inline docs/contribution/CONTRIBUTION.md:1]: Hi @josephpdf, please update this file CONTRIBUTION.md by adding the content/file from: [Contribution Guidelines](https://github.com/ByteBuildersLabs/ByteBeastsBackend/blob/main/CONTRIBUTION.md) Thanks! 🫡

File: .github/templates/PULL_REQUEST_TEMPLATE.md
Comment (coxmars) [inline .github/templates/PULL_REQUEST_TEMPLATE.md:1]: Hi @josephpdf, please use the following structure for this Pull Request Template:

## Pull Request Overview
### 📝 Summary 
- Closes # number issue
- Brief summary
### 🔄 Changes Made
- Add changes made
### 🔧 Tests Results
- Add images to validate your results

**Review by danielcdz — APPROVED** [review]
LGTM

**Review by coxmars — APPROVED** [review]
LGTM

**Pull request merged**